Managers/MapManager.gd

```diff
@@ -195,13 +197,35 @@ func assign_events(map: MapBase = current_map) -> void:
 
 			else:
 				var room_type: GlobalEnums.EventType
-				while not is_room_event_correct(_current_room, map):
+				map_reset_counter = 0
+				while not is_room_event_correct(_current_room, map) and map_reset_counter < map_reset_limit:
+					map_reset_counter += 1
 					room_type = pick_room_type()
 					_current_room.room_event = GlobalEnums.choose_event_from_type(room_type)
+
+				# If we couldn't find a suitable room type in map_reset_limit tries, we regenerate the map
+				if map_reset_counter >= map_reset_limit:
+					map_reset_flag = true
+					break
 				
-				# We remove the event from the list only after we are sure it does not conflict with any rules
+				# If assigned, we remove the event from the list only after we are sure it does not conflict with any rules
 				event_list.erase(room_type)
 
+		if map_reset_flag:
+			break
+
+
+	if map_reset_flag:
+		# Reset all room events to null
+		for index_height: int in range(map.rooms.size()):
+			for index_width: int in range(map.rooms[index_height].size()):
+				if map.rooms[index_height][index_width] == null:
+					continue
+
+				map.rooms[index_height][index_width].room_event = null
```

Comment:
instead of having this double break, can you not put the reset of room events after the `map_reset_counter >= map_reset_limit` check ?

Reply:
I thought of it but if we reset in the for loops, after the function call ends wouldn't it resume the for loops?

Reply:
can you not do the reset then call the `generate_event` in the block ? that would recurse
alternatively, put all the room at null when you call "generate event" and just call `generate_event` if need to regenerate

Reply:
I'm not sure to understand your reasoning, but the "problem" is the `generate_event` call itself:

![image](https://github.com/user-attachments/assets/2af0515e-3475-4e1e-9d92-11bd6a2c35a2)

If we call `generate_event` inside the for loops, doesn't it resume after the call ends like on this image? And if there are multiple recursive calls (which are very rare of course), it would then need to finish the previous loops. And with a bigger map, it could be slower

My point with the breaks are just to avoid that, like on this image:

![image](https://github.com/user-attachments/assets/2cf4f634-bcf1-4b52-83c3-a647f5d717dd)

Tell me if it's unclear.

Reply:
![image](https://github.com/user-attachments/assets/2854a815-293e-469c-ba51-7b80e7b7a6ed)
Example of the recursive calls, I think it happens like that (?) if we don't break out of the for loops

Reply:
Hm, while it doesn't become super slow or anything, I thought breaks would be better imo

Reply:
solution: instead of a `break` in the for loop after the recursive call to `regenerate_map`, use a `return`. The problem in the current for loop is that since there are two levels loop, indeed a break would continue the outer loop, which wouldn't happen with a return. So this:
```GDScript
if map_reset_counter >= map_reset_limit:
	# Reset all room events to null
	for index_height: int in range(map.rooms.size()):
		for index_width: int in range(map.rooms[index_height].size()):
			if map.rooms[index_height][index_width] == null:
				continue

			map.rooms[index_height][index_width].room_event = null
	assign_event(map)
	return # since we return here, we do not continue the for loop we were in
```

Reply:
I just thought the double break could lead to confusion later on

Reply:
Ah right it's much simpler